File: 3702-longest-subsequence-with-non-zero-bitwise-xor/3702-longest-subsequence-with-non-zero-bitwise-xor.c
int longestSubsequence(int* nums, int numsSize) {
    int result = 0;
    bool iszero = true;
    for(int i = 0; i < numsSize ; i++){
        if(nums[i] != 0) iszero = false;
        result ^= nums[i];
        
    }

    if(iszero){
        return 0;
    }

    if(result != 0){
        return numsSize;
    }else{
        return numsSize - 1;
    }
    return 0;
}

File: 3702-longest-subsequence-with-non-zero-bitwise-xor/3702-longest-subsequence-with-non-zero-bitwise-xor.py
class Solution:
    def longestSubsequence(self, nums: List[int]) -> int:
        result = 0
        allzero = True
        for i in range(len(nums)):
            if nums[i] != 0:
                allzero = False
            result ^= nums[i]
        if allzero == True:
            return 0
        if result != 0:
            return len(nums)
        else:
            return len(nums) - 1